### PR TITLE
Mutate block-pass arguments

### PR DESCRIPTION
**Pull request by dgollahon**
- These were previously not recursing and being mutated.
- Remove Noop mutator--this behavior (AFAICT) is equivalent to just
  being a node in the generic list. The `cbase` node still gets deleted.

### CODE REVIEW

File: meta/block_pass.rb [hunk 1]
```diff
@@ -3,4 +3,6 @@
 
   singleton_mutations
   mutation 'foo'
+  mutation 'foo(&nil)'
```
Comment (backus) [inline meta/block_pass.rb]: Any way we can avoid emitting these mutations? They are invalid.

**Comment by backus** [comment]
Can you add a test for `foo(&method(:bar))`?

File: meta/block_pass.rb (hunk 1 above)
Comment (dgollahon) [inline meta/block_pass.rb]: They are not invalid. I had a check for them, but I tried it, and you can actually do `&nil`. It's a no-op so we could choose not to do it if we wanted. We could also skip `&self` but that works for objects that define `#to_proc`.

```ruby
[1].map(&nil).to_a # => [1]
```

Comment (backus) [inline meta/block_pass.rb]: Oh cool. Alrighty then

**Comment by dgollahon** [comment]
> Can you add a test for foo(&method(:bar))?

I switched the examples to just use the `method` example.

**Comment by backus** [comment]
Looks good to me!

**Pull request merged**